x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/find_rules_route.ts

```diff
@@ -59,45 +62,94 @@ export const findRulesRoute = (router: SecuritySolutionPluginRouter) => {
           fields: query.fields,
         });
 
+        if (rules.data.length === 0) {
+          const [validated, errors] = transformValidateFindAlerts(rules, [], []);
+          if (errors != null) {
+            return siemResponse.error({ statusCode: 500, body: errors });
+          } else {
+            return response.ok({ body: validated ?? {} });
+          }
+        }
+
+        const ruleIds = rules.data.map((rule) => rule.id);
+
+        // // if any rules attempted to execute but failed before the rule executor is called,
+        // // an execution status will be written directly onto the rule via the kibana alerting framework,
+        // // which we are filtering on and will write a failure status
+        // // for any rules found to be in a failing state into our rule status saved objects
+        // const failingRules = rules.data.filter(
+        //   (rule) => rule.executionStatus != null && rule.executionStatus.status === 'error'
+        // );
+
+        // const ruleStatuses = await Promise.all(
+        //   rules.data.map(async (rule) => {
+        //     const results = await ruleStatusClient.find({
+        //       perPage: 1,
+        //       sortField: 'statusDate',
+        //       sortOrder: 'desc',
+        //       search: rule.id,
+        //       searchFields: ['alertId'],
+        //     });
+        //     const failingRule = failingRules.find((badRule) => badRule.id === rule.id);
+        //     if (failingRule != null) {
+        //       if (results.saved_objects.length > 0) {
+        //         results.saved_objects[0].attributes.status = 'failed';
+        //         results.saved_objects[0].attributes.lastFailureAt = failingRule.executionStatus.lastExecutionDate.toISOString();
+        //       }
+        //     }
+        //     return results;
+        //   })
+        // );
+
+        // const ruleActions = await Promise.all(
+        //   rules.data.map(async (rule) => {
+        //     const results = await getRuleActionsSavedObject({
+        //       savedObjectsClient,
+        //       ruleAlertId: rule.id,
+        //     });
+
+        //     return results;
+        //   })
+        // );
+
+        const [ruleActions, ruleStatuses] = await Promise.all([
+          bulkFetchRuleActionsOfManyRules({ savedObjectsClient, ruleIds }),
+          bulkFetchLatestStatusesOfManyRules(elasticsearchClient.asCurrentUser, ruleIds),
+        ]);
```

Comment:
Please review this from the security standpoint (access privileges, isolation (spaces), error handling, potential issues, etc.).

- Is it ok to query `.kibana` index via `elasticsearchClient.asCurrentUser`, or privileges to `.kibana` are not guaranteed to a typical user of security solution?
- I guess it would be better to query `asInternalUser`. Would it be safe in this particular case?